src/subplots.ts

```diff
@@ -1534,6 +1547,14 @@ export class BasePlot extends PlotData {
 
   get falseIndicesArray(): boolean[] { return new Array(this.nSamples).fill(false) }
 
+  get offset_factor(): Vertex { return this._offset_factor ?? new Vertex(0.035, 0.07) }
```

Comment:
Yes they are since default values are not the same in parent and children

Reply:
Why can't the children just redefine the default value to its own ?

Reply:
I am also questioning about that.

But what is clear is that you cannot expect this to work as expected:
```
class Parent {
    attribute: type = X;
    constructor() {
       this.function(attribute)
    }
}

class Children extends Parent {
    attribute: type = Y;
    constructor() {
       super()
    }
}
```

Here, the used value of attribute in children **when running the constructor** (which is what you want to do) will be X and not Y; because at this moment in the run of constructor, you are still in Parent class, although you are creating a Children instance.

To avoid this behavior, the Internet gives us the above trick.

Reply:
https://www.typescriptlang.org/play?#code/MYGwhgzhAECC0G8BQ1XQA4FcBGICWwGATgPboBc0EALkXgHYDm0AvNAERjsppa4HRGjSmHoBPANw9UwEvRpFMwaiSIAKAJSJpaVNQAWeCADoVKtQaPH0pdBp0BfHX3yEzJNTbJbku3ZZMhVmIyKV0nJyRQSBgAIWgAUwAPagT6ABMYeF9eHFcQiipaBmY2dmxuXVl5WiUVdR8dXQhMdASGxyRI6ppoMGD6BIB3OE0pHupobAHh6FixpCi5CBIQBOMQEkY1dnhKdgAaPuMhI-ZYw6mTxnsgA

Reply:
Ok, I think this case made me understand what is the difference between inside/outside attribute assignment in js, which was not clear at all to me before.

They are actually not equivalent, after all. I think on the other side, that they are a bit similar to python's class attribute vs attribute, even if they are not the same.

Considering [this thread](https://stackoverflow.com/questions/64436532/javascript-class-property-inside-vs-outside-constructor), class fields (outside the constructor), are called **after** the constructor which leads to the shown behavior.

Actually, the `prop` attribute in our test case, is not static nor a class field, its a plain "instance attribute". In this case, I think we should then use the constructor syntax : 

https://www.typescriptlang.org/play?#code/MYGwhgzhAECC0G8BQ1XQA4FcBGICWw0A5kQFzRgB2AngNwprAD2lEALgE6bBtMcAUDNGiy4CGDk3Tl2HPJSLQAvNABEYVUOgBKRFuFsAFnggA6dJPTKJU+sIPGzvXv21aAvltH5Czpq717NCMTUxJrELMLWw8kTyRQSBgAIWgAUwAPNjTKABMYeGRhZlZObl4BfREcHxtpaFl5RRVVbE1hXSKghsx0NIFo9DdhT3iS9gprSjSAdzhXenG2aGwp2ehkhaQElggmEDTTECYiflV4clUAGgowohvVZOuVu7cgA